src/game/Spells/Spell.cpp

```diff
@@ -5951,6 +5943,19 @@ SpellCastResult Spell::CheckCast(bool strict)
                 return SPELL_FAILED_BAD_TARGETS;
             break;
         }
+        case 27230: // Health Stone
+        case 11730:
+        case 11729:
+        case 6202:
+        case 6201:
+        case 5699:
+        {
+            // check if we already have a healthstone
+            uint32 itemType = GetUsableHealthStoneItemType(m_caster);
+            if (itemType && m_caster->IsPlayer() && ((Player*)m_caster)->GetItemCount(itemType) > 0)
+                return SPELL_FAILED_TOO_MANY_OF_ITEM;
+            break;
+        }
```

Comment:
@killerwife @BenDol 
Maybe move this under this line? https://github.com/cmangos/mangos-tbc/blob/80d2abeb9aa7c5aedff744b8568231feb296bcdb/src/game/Spells/Spell.cpp#L4652
```c++
case SPELLFAMILY_WARLOCK:
  if( m_spellInfo->SpellFamilyFlags & uint64(0x0000000000100000) ){
    
    // check if we already have a healthstone
    uint32 itemType = GetUsableHealthStoneItemType(m_caster);
    if (itemType && m_caster->IsPlayer() && ((Player*)m_caster)->GetItemCount(itemType) > 0)
      return SPELL_FAILED_TOO_MANY_OF_ITEM;
  }
  break;
```

Reply:
I disagree with the mask approach, because then it wont be transferrable to spell scripting easily. It will have to be assigned to specific spell IDs.

Reply:
@killerwife i don't understand you at all... 😞 m_spellInfo already contains Id inside anyway. And instead 6 checks it will check only once. This is just a check optimization speed.
And how this check can be related to spell scripting? And more bigger question, about what spell scripting are you talking about? I feel myself dumb right now...

Reply:
The scripting that is not implemented yet. I am merely thinking 6 months ahead. I am sure stuff will be making more sense soon™.